src/common/webpack/config.ts

```diff
@@ -461,7 +463,70 @@ function createJavaScriptLoader({
     config,
     isSsr,
 }: HelperOptions): webpack.RuleSetUseItem {
+    if (config.javaScriptLoader === 'swc') {
+        if (config.bundler !== 'rspack') {
+            throw new Error('SWC is supported only with rspack bundler');
+        }
+
+        const browserslist = require('browserslist');
+
+        const options: SwcLoaderOptions = {
+            module: {
+                type: 'es6',
+            },
+            env: {
+                targets: isSsr ? {node: 'current'} : browserslist(),
+                mode: 'usage',
+                coreJs: '3.40.0',
+                bugfixes: true,
+            },
+            isModule: 'unknown',
+            jsc: {
+                parser: {
+                    syntax: 'typescript',
+                    tsx: true,
+                },
+                transform: {
+                    react: {
+                        runtime: config.newJsxTransform ? 'automatic' : 'classic',
+                        development: isEnvDevelopment,
+                        refresh: !isSsr && isEnvDevelopment && config.reactRefresh !== false,
+                        useBuiltins: true,
+                    },
+                    optimizer: isEnvProduction
+                        ? {
+                              simplify: true,
+                              jsonify: {minCost: 1024},
+                          }
+                        : undefined,
+                },
+                assumptions: {
+                    privateFieldsAsProperties: true,
+                    setPublicClassFields: true,
+                },
+            },
+            rspackExperiments:
+                !isSsr && isEnvProduction
+                    ? {
+                          import: [
+                              {
+                                  libraryName: 'lodash',
+                                  customName: 'lodash/{{ member }}',
+                              },
+                          ],
+                      }
+                    : undefined,
```

Comment:
```suggestion
                experimental: {
                    plugins: [
                        !isSsr && isEnvProduction
                            ? ([
                                  require.resolve('@swc/plugin-transform-imports'),
                                  {
                                      lodash: {
                                          transform: 'lodash/{{member}}',
                                      },
                                  },
                              ] satisfies [string, Record<string, any>])
                            : undefined,
                    ].filter((p) => p !== undefined),
                },
            },
```

Reply:
I think that using a built-in rspack plugin is better than custom

Reply:
This is a plugin from swc and will work for both loaders.